3MI0800270/2023-03-07/zad5.cpp

```diff
@@ -0,0 +1,335 @@
+    }
+    file.close();
+    initUser(users[size], user.username, user.password, user.email, filepath);
+    return writeUsers(users, size + 1, filepath);
+}
+
+bool removeUser(User user, const char* filepath) {
+    int size = getTableSize(filepath);
+    User* users = new User[size];
+    ifstream file(filepath);
+    int index = 0;
+    for (int i = 0;i < size;i++) {
+        readUser(users[i], file);
+        //cout << users[i].username << " " << user.username << endl;
+        if (strcmp(users[i].username, user.username) == 0) {
+            cout << users[i].username << endl;
+            index = i;
+        }
+    }
+    swap(users[index], users[size - 1]);
+    file.close();
+    return writeUsers(users, size - 1, filepath);
+}
+
+bool registerStart(User& user) {
+    User temp;
+    while (true) {
+        cout << "Please enter your username, password and email\n- потребителско име - дължина до 32 символа, може да съдържа букви, цифри, -, _, .\n- парола - най - малко 8 символа, най - много 32, трябва да съдържа поне една главна буква, малка буква и цифра\n- имейл адрес - дължина до 32 символа" << endl;
+        char username[33];
+        char password[33];
+        char email[33];
+        cin >> username >> password >> email;
+
+        if (initUser(temp, username, password, email, FILE_PATH)) {
+            if (addUser(temp, FILE_PATH)) {
+                cout << "Welcome" << endl;
+                initUser(user, temp.username, temp.password, temp.email);
+                return 1;
+            }
+            else {
+                cout << "DB is full" << endl;
+                return 0;
+            }
+        }
+        else {
+            cout << "Your information does not fulfill the requirments, try again." << endl;
+        }
+    }
+}
+
+bool loginStart(User& user) {
+    int size = getTableSize(FILE_PATH);
+    ifstream file(FILE_PATH);
+    while (true)
+    {
+        cout << "Please enter your username and password" << endl;
+        char username[33];
+        char password[33];
+        cin >> username >> password;
+        for (int i = 0;i < size;i++) {
+            User temp;
+            readUser(temp, file);
+            if (!strcmp(username, temp.username) && !strcmp(username, temp.username)) {
+                file.close();
+                cout << "Login Successful" << endl;
+                initUser(user, temp.username, temp.password, temp.email);
+                return 1;
+            }
+        }
+        cout << "Login Unsuccessful. Try again." << endl;
+    }
+
+}
+
+void changePassword(User& user) {
+    while (true) {
+        cout << "Enter new password: ";
+        char pass[33];
+        cin >> pass;
+        if (validatePassword(pass)) {
+            initUser(user, user.username, pass, user.email);
+            removeUser(user, FILE_PATH);
+            addUser(user, FILE_PATH);
+            cout << "Password changed." << endl;
+            break;
+        }
+    }
+}
+
+void cliStart() {
```

Comment:
btw - you may enjoy [this read](https://refactoring.guru/design-patterns/command) :P